api/config/v1alpha1/validation/validate.go

```diff
@@ -93,6 +96,18 @@ func validateServiceType(spec *egcfgv1a1.EnvoyProxySpec) []error {
 	return errs
 }
 
+func validateServiceAllocateLoadBalancerNodePorts(spec *egcfgv1a1.EnvoyProxySpec) error {
```

Comment:
thoughts on flattening this and above `validateServiceType` into a single function called `validateService` ?

Reply:
Sounds good to me. I will update it